app/Middleware/Global/AccessAuth.ts

```diff
@@ -20,26 +20,21 @@ export default class AccessAuth {
         required: 'You must provide an access_id header to access the api. Read more on ...',
       },
     })
-    try {
-      const access = await Access.findByOrFail('access_id', accessId)
-      if (access.type === 'token') {
-        if (!accessToken) {
-          throw new Error(
-            'You must provide an access_token header to access the api. Read more on ...'
-          )
-        } else if (access.token !== accessToken) {
-          throw new Error('Invalid access token')
-        }
-      } else {
-        const origin = request.header('origin')
-        if (!origin || !access.domains!.includes(origin)) {
-          throw new Error('Access denied from this domain')
-        }
+    const access = await Access.findByOrFail('access_id', accessId)
```

Comment:
```suggestion
    const access = await Access.findBy('access_id', accessId)
    if (!access) {
      throw new Error('Invalid access id')
    }
```

Reply:
I must check, I think I had issues doing this.

Reply:
It's better to have a custom error message instead of the default `ROW_NOT_FOUND` error

Reply:
btw all errors should be replaced with `response.unauthorized` to get a correct status code....
I think that `throw ...` gives a 500 - Internal server error whereas the 401 - Unauthorized status code would better match the error message.

Reply:
We should maybe work on a utility function for error handling